tests/tests_pytorch/trainer/logging_/test_logger_connector.py

```diff
@@ -639,3 +639,23 @@ def test_result_collection_no_batch_size_extraction():
     assert results["training_step.epoch_log_val"].value == log_val * batch_size
     assert results["training_step.epoch_log_val"].cumulated_batch_size == batch_size
     assert results["training_step.epoch_sum_log_val"].value == log_val
+
+
+def test_result_collection_changes_device():  # mock_torch):
```

Comment:
@clumsy Let's upgrade this test to use the real device and run on the GPU? You can add

```suggestion
@RunIf(min_cuda_gpus=1)
def test_result_collection_changes_device():  # mock_torch):
```

and replace the mocks with cpu and cuda devices?

Reply:
But why? There's no logic that is specific to either CPU or GPU here. In `teardown()` the metrics are moved to CPU unconditionally. I think it's perfectly unit-testable with mocks whether it's run on CPU, GPU or even XLA.
I guess I can make an extra test just for GPU in addition to this unit test if you feel strongly about this.

Reply:
No need for an extra test. We can certainly keep it a unit test, I didn't mean to change that. All I suggest is to change the mock device into `torch.device("cpu")` and `torch.device("cuda")`. Having both would not add any value. We also don't need to test with all devices, because the logic we're testing is not about a concrete device, but about it changing to the target device when moving the result object.